lib/i18n/tasks/missing_keys.rb

```diff
@@ -35,36 +35,48 @@ def eq_base_keys(opts = {})
       end
     end
 
-    def missing_diff_forest(locales, base = base_locale)
+    def missing_diff_forest(locales, base = base_locale, skip_interpolation = false)
       tree = empty_forest
       # present in base but not locale
       (locales - [base]).each do |locale|
-        tree.merge! missing_diff_tree(locale, base)
+        tree.merge! missing_diff_tree(locale, base, skip_interpolation)
       end
       if locales.include?(base)
         # present in locale but not base
         (self.locales - [base]).each do |locale|
-          tree.merge! missing_diff_tree(base, locale)
+          tree.merge! missing_diff_tree(base, locale, skip_interpolation)
         end
       end
       tree
     end
 
-    def missing_used_forest(locales, _base = base_locale)
+    def missing_used_forest(locales, _base = base_locale, _skip_interpolation = false)
       locales.inject(empty_forest) do |forest, locale|
         forest.merge! missing_used_tree(locale)
       end
     end
 
-    def missing_plural_forest(locales, _base = base_locale)
+    def missing_plural_forest(locales, base = base_locale, _skip_interpolation = false)
       locales.each_with_object(empty_forest) do |locale, forest|
         required_keys = required_plural_keys_for_locale(locale)
         next if required_keys.empty?
 
         tree = empty_forest
-        plural_nodes data[locale] do |node|
+        plural_nodes data[base] do |node|
           children = node.children
-          present_keys = Set.new(children.map { |c| c.key.to_sym })
+
+          # Get the existing translated node if it exists
+          translated_node = data[locale].get("#{locale}." + node.full_key(root: false))
```

Comment:
I think this is the same as:

```suggestion
          translated_node = data[locale].get(node.full_key(root: true))
```

Reply:
Let me double check but I think the goal here is to to copy the value from the locale we're iterating over from the passed locales in L60 (if it exists). If we used the node value, it'd give us the value from the base locale each time.

Reply:
Ah, that makes sense

Reply:
[f655f32](https://github.com/glebm/i18n-tasks/pull/596/commits/f655f32095f8c9c34c4327e769dfae67234527f4)